src/main/java/com/dataguise/saas/controllers/SettingsController.java

```diff
@@ -0,0 +1,134 @@
+            }
+    }
+
+    @RequestMapping(value = "/loadSettings", method = RequestMethod.GET)
+    public SettingsDTO loadSettings() throws BadGatewayException {
+
+        SettingsDTO settingsDTO = new SettingsDTO();
+        List<String> policyList = new ArrayList<>();
+        List<DgPreferenceStruct> policyPrefList = null;
+
+        settingsDTO.setDgsecureClassicURL(getDgSecureURL());
+
+        try {
+            policyPrefList = driver.getDefaultPolicy();
+        } catch (DgDrvException e) {
+            throw new BadGatewayException("Unable to load policy value.", HttpStatus.BAD_GATEWAY);
+        }
+        if(!policyPrefList.isEmpty()) {
+            String policy = policyPrefList.get(0).getPrefParamValue();
+            if(policy.equals("") || policy == null) {
+                throw new BadGatewayException("Policy value is not set.", HttpStatus.BAD_GATEWAY);
+            }
+            if(policy.contains(",")) {
+                String[] policies = policy.split(",");
+                for(String s : policies) {
+                    policyList.add(s);
+                }
+            } else {
+                policyList.add(policy);
+            }
+        } else {
+            logger.debug("Empty policy pref list from DgSecure. Is SaaS.UI.Default_Policy supported by DgSecure version?");
+        }
+        settingsDTO.setPolicyType(policyList);
+
+        return settingsDTO;
+    }
+
+
+    public String getDgSecureURL() {
+        DgSecureDetails dgSecure = new DgSecureDetails();
+        URL controllerBaseURL = null;
+        try {
+            controllerBaseURL = new URL(driver.getControllerClient().getApiClient().getBasePath());
+        } catch (MalformedURLException e) {
+            logger.error("Malformed DgSecure url.");
+        }
+        if(controllerBaseURL.getProtocol().equalsIgnoreCase("http")){
+            dgSecure.setSecure(false);
+        }else if(controllerBaseURL.getProtocol().equalsIgnoreCase("https")){
+            dgSecure.setSecure(true);
+        }
+        dgSecure.setHost(controllerBaseURL.getHost());
+        dgSecure.setPort(controllerBaseURL.getPort());
+        dgSecure.setProtocol(controllerBaseURL.getProtocol());
+        dgSecure.setDgsecureURL(String.format("%s://%s:%s/",controllerBaseURL.getProtocol(), controllerBaseURL.getHost(), controllerBaseURL.getPort()));
+        return dgSecure.getDgsecureURL();
+    }
+
+    public String getJarPath(){
+        String jarPath = SettingsController.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+        try {
+            jarPath = URLDecoder.decode(jarPath,"utf-8");
+            jarPath = jarPath.substring(0, jarPath.indexOf("!"));
+            jarPath = jarPath.substring(0,jarPath.lastIndexOf("/"));
+            jarPath = jarPath.substring(jarPath.indexOf(":")+1,jarPath.length());
+            logger.info("jarPath is: "+jarPath );
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/controllers/SettingsController.java L127
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1792676523>*